x-pack/plugins/cases/public/containers/use_get_feature_ids.tsx

```diff
@@ -5,70 +5,36 @@
  * 2.0.
  */
 
-import { useCallback, useEffect, useState, useRef } from 'react';
+import { useQuery } from '@tanstack/react-query';
 import type { ValidFeatureId } from '@kbn/rule-data-utils';
-
-import * as i18n from './translations';
+import type { ServerError } from '../types';
 import { useToasts } from '../common/lib/kibana';
+import * as i18n from './translations';
 import { getFeatureIds } from './api';
+import { casesQueriesKeys } from './constants';
 
-const initialStatus = {
-  isLoading: true,
-  alertFeatureIds: [] as ValidFeatureId[],
-  isError: false,
-};
-
-export const useGetFeatureIds = (
-  alertRegistrationContexts: string[]
-): {
-  isLoading: boolean;
-  isError: boolean;
-  alertFeatureIds: ValidFeatureId[];
-} => {
+export const useGetFeatureIds = (alertRegistrationContexts: string[]) => {
   const toasts = useToasts();
-  const isCancelledRef = useRef(false);
-  const abortCtrlRef = useRef(new AbortController());
-  const [status, setStatus] = useState(initialStatus);
-
-  const fetchFeatureIds = useCallback(
-    async (registrationContext: string[]) => {
-      setStatus({ isLoading: true, alertFeatureIds: [], isError: false });
-      try {
-        isCancelledRef.current = false;
-        abortCtrlRef.current.abort();
-        abortCtrlRef.current = new AbortController();
-
-        const query = { registrationContext };
-        const response = await getFeatureIds(query, abortCtrlRef.current.signal);
-
-        if (!isCancelledRef.current) {
-          setStatus({ isLoading: false, alertFeatureIds: response, isError: false });
-        }
-      } catch (error) {
-        if (!isCancelledRef.current) {
-          setStatus({ isLoading: false, alertFeatureIds: [], isError: true });
-          if (error.name !== 'AbortError') {
-            toasts.addError(
-              error.body && error.body.message ? new Error(error.body.message) : error,
-              { title: i18n.ERROR_TITLE }
-            );
-          }
-        }
-      }
+  return useQuery<ValidFeatureId[], ServerError>(
+    casesQueriesKeys.featureIds(alertRegistrationContexts),
+    () => {
+      const abortCtrlRef = new AbortController();
+      const query = { registrationContext: alertRegistrationContexts };
+      return getFeatureIds(query, abortCtrlRef.signal);
     },
-    [toasts]
+    {
+      onError: (error: ServerError) => {
+        if (error.name !== 'AbortError') {
```

Comment:
You can use 

```
const { showErrorToast, showSuccessToast } = useCasesToast();


showErrorToast(error, { title: i18n.ERROR_TITLE });
```

Check `x-pack/plugins/cases/public/containers/use_get_cases_status.tsx` for example.

Reply:
Fixed

Reply:
There is no need for the `if (error.name !== 'AbortError') {`. The `showErrorToast` will take care of that for you.